cranelift/codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -1388,41 +1396,44 @@ pub(crate) fn define<'shared>(
             ),
     );
 
-    // Stack addresses.
+    // Stack accesses.
     //
     // TODO Alternative forms for 8-bit immediates, when applicable.
 
-    recipes.add_template_recipe(
-        EncodingRecipeBuilder::new("spaddr4_id", &formats.stack_load, 6)
+    recipes.add_template(Template::new(
+        EncodingRecipeBuilder::new("spaddr_ld_id", &formats.stack_load, 6)
             .operands_out(vec![gpr])
             .emit(
                 r#"
-                    let sp = StackRef::sp(stack_slot, &func.stack_slots);
-                    let base = stk_base(sp.base);
-                    {{PUT_OP}}(bits, rex2(out_reg0, base), sink);
-                    modrm_sib_disp8(out_reg0, sink);
-                    sib_noindex(base, sink);
-                    let imm : i32 = offset.into();
-                    sink.put4(sp.offset.checked_add(imm).unwrap() as u32);
-                "#,
+                        let sp = StackRef::sp(stack_slot, &func.stack_slots);
```

Comment:
I know you and Benjamin discussed this a bit in the earlier version of this PR, but is there a downside to reporting a possible stack overflow on `stack_load`? Just the number of offset/`TrapCode` pairs reported for the function?

I'd say most `spst_id` `StackOverflow` are impossible as well. Even so, it doesn't seem bad to have the second trap site for each spill. If the number of trap sites ends up a concern we'll probably want to do something a bit more clever to pare down trap sites anyway. Reporting the trap site avoids a silent error (missing trap site) in the face of (maybe?) misuse.